numpy/core/src/umath/ufunc_object.c

```diff
@@ -2844,7 +2843,9 @@ PyUFunc_Reduce(PyUFuncObject *ufunc, PyArrayObject *arr, PyArrayObject *out,
     }
 
     /* Get the identity */
-    identity = _get_identity(ufunc, &reorderable);
+    if (identity == NULL || identity == Py_None) {
+        identity = _get_identity(ufunc, &reorderable);
+    }
```

Comment:
You need an `else { Py_INCREF(identity); }`, so that the decref at the end is safe.

Reply:
I don't think so, it only checks for `None` and `NULL`. The object is getting passed through.

`identity` is gotten from the `PyArg_ParseTupleAndKeywords`, does this do the `Py_INCREF` call automatically or not?

Reply:
And the other half of the puzzle is that `_get_identity` _does_ increment the reference count - so you want to make them equivalent so the cleanup doesn't need special casing.